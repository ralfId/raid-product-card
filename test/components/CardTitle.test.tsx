import React from 'react';
import renderer from 'react-test-renderer';
import { CardTitle, ShoppingCard } from '../../src/components';
import { product1 } from '../data/data';

describe('card title', () => {
    test('debe mostrar el titulo de la card correctamente', () => {
        const wrapper = renderer.create(<CardTitle title="Titulo del producto" className='custom-class' />);

        //console.log(wrapper.toJSON());
        expect(wrapper.toJSON()).toMatchSnapshot();
    })
})


describe('ShoppingCard with title', () => {
    test('debe mostrar la tarjeta con el nombre del producto', () => {
        const wrapper = renderer.create(
            <ShoppingCard product={product1}>
                {
                    () => (

                        <CardTitle />
                    )
                }
            </ShoppingCard>
        );

        //console.log(wrapper.toJSON());
        expect(wrapper.toJSON()).toMatchSnapshot();
    })
})