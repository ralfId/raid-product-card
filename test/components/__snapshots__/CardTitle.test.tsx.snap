// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCard with title debe mostrar la tarjeta con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    coffee card 1
  </span>
</div>
`;

exports[`card title debe mostrar el titulo de la card correctamente 1`] = `
<span
  className="productDescription custom-class"
>
  Titulo del producto
</span>
`;
